src/pages/post-test-summary/post-test-summary.ts

```diff
@@ -72,7 +79,33 @@ export class PostTestSummaryPage {
     showMeQuestionModal.present();
   };
 
-  backToJournal() {
+  onSubmitPress() {
+    if (this.isFullyComplete()) {
+      this.backToJournal();
+    } else {
+      const title = 'Defer test result submission?';
+      const message =
+        'You must complete all the mandatory fields before this test result can be submitted.' +
+        ' Are you sure you want to defer submission to a later time?';
+      this.alertCtrl
+        .create({
+          title,
+          message,
+          buttons: [
+            {
+              text: 'Cancel'
+            },
+            {
+              text: 'Defer',
+              handler: () => this.backToJournal()
```

Comment:
Do we need this additional wrapper? can we just pass `this.backToJournal` instead?

Reply:
I tried without the wrapping arrow, I had a `this` binding issue.

Reply:
fair enough, missed that this was all happening in a`.create`